instrumentation/graphql/test/instrumentation/graphql/tracers/graphql_tracer_test.rb

```diff
@@ -0,0 +1,292 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+require_relative '../../../../lib/opentelemetry/instrumentation/graphql'
+require_relative '../../../../lib/opentelemetry/instrumentation/graphql/tracers/graphql_tracer'
+
+describe OpenTelemetry::Instrumentation::GraphQL::Tracers::GraphQLTracer do
+  let(:instrumentation) { OpenTelemetry::Instrumentation::GraphQL::Instrumentation.instance }
+  let(:exporter) { EXPORTER }
+  let(:spans) { exporter.finished_spans }
+  let(:config) { {} }
+
+  let(:query_string) do
+    <<-GRAPHQL
+      query($id: Int!){
+        simpleField
+        resolvedField(id: $id) {
+          originalValue
+          uppercasedValue
+        }
+      }
+    GRAPHQL
+  end
+
+  before do
+    # Reset various instance variables to clear state between tests
+    [GraphQL::Schema, SomeOtherGraphQLAppSchema, SomeGraphQLAppSchema].each(&:_reset_tracer_for_testing)
+    instrumentation.instance_variable_set(:@installed, false)
+
+    # This test file is all about testing the GraphQLTracer class
+    # so we're always going to force legacy_tracing to be true
+    config[:legacy_tracing] = true
+    instrumentation.install(config)
+
+    exporter.reset
+  end
+
+  if OpenTelemetry::Instrumentation::GraphQL::Instrumentation.instance.supports_legacy_tracer?
```

Comment:
In some tests you are skipping explicitly checking this in individual tests, but in this case you are omitting test definitions altogether.

Why is this taking a different approach from other tests?

Reply:
The install tests are bit of `if this version then that should install` `if this version should fail to install` where as the GraphQLTracer and GraphQLTrace test suites should only run if they are being tested against a supported GraphQL version. 

It felt like it made sense but if this is coming as question, maybe it doesn't? Would it make more sense to have every single test in the GraphQLTrace(r) files check if its a supported version?

Reply:
Yes I think so. 

I wonder if this kind of duality merits multiple test suites where we check for version in the test runner instead of the classes themselves? IDK

Reply:
Would it be clearer to `skip unless OpenTelemetry::Instrumentation::GraphQL::Instrumentation.instance.supports_legacy_tracer?` in the before block?